lib/includes.js

```diff
@@ -144,12 +144,25 @@ module.exports = function (adapter, schemas) {
             fetchedIds[type] = fetchedIds[type].concat(remainingIds);
 
             if (!inclusionDescriptor.def.baseUri) {
-                return adapter.find(type, {query: {filter: {id: remainingIds.join(',')}}});
+                return new Promise(function (resolve, reject) {
+                    const url = '/' + type + '?filter[id]=' + remainingIds.join(',');
+                    server.inject({method: 'get', url: url, headers: req.headers}, function (res) {
+                        if (res.statusCode !== 200) {
+                            reject(new Error(JSON.stringify(res.result)))
```

Comment:
@mavdi 400 is for user errors. The internal state of DB is rather server side error (500).